src/coreclr/src/inc/eventtracebase.h

```diff
@@ -333,44 +345,61 @@ class XplatEventLoggerConfiguration
         return ComponentSpan(start, end);
     }
 
-    LPCWSTR ParseProviderName(ComponentSpan const & component) const
+    NewArrayHolder<WCHAR> ParseProviderName(ComponentSpan const & component) const
     {
-        auto providerName = (WCHAR*)nullptr;
+        NewArrayHolder<WCHAR> providerName = nullptr;
         if ((component.End - component.Start) != 0)
         {
             auto const length = component.End - component.Start;
             providerName = new WCHAR[length + 1];
-            memset(providerName, '\0', (length + 1) * sizeof(WCHAR));
             wcsncpy(providerName, component.Start, length);
+            providerName[length] = '\0';
         }
         return providerName;
     }
 
-    ULONGLONG ParseEnabledKeywordsMask(ComponentSpan const & component) const
+    uint64_t ParseEnabledKeywordsMask(ComponentSpan const & component) const
     {
-        auto enabledKeywordsMask = (ULONGLONG)(-1);
+        auto enabledKeywordsMask = (uint64_t)(-1);
         if ((component.End - component.Start) != 0)
         {
             enabledKeywordsMask = _wcstoui64(component.Start, nullptr, 16);
         }
         return enabledKeywordsMask;
     }
 
-    UINT ParseLevel(ComponentSpan const & component) const
+    uint32_t ParseLevel(ComponentSpan const & component) const
     {
-        auto level = TRACE_LEVEL_VERBOSE;
+        int level = TRACE_LEVEL_VERBOSE; // Verbose
         if ((component.End - component.Start) != 0)
         {
             level = _wtoi(component.Start);
         }
         return level;
     }
 
-    LPCWSTR _provider;
-    ULONGLONG _enabledKeywords;
-    UINT _level;
+    NewArrayHolder<WCHAR> ParseArgument(ComponentSpan const & component) const
+    {
+        NewArrayHolder<WCHAR> argument = nullptr;
+        if ((component.End - component.Start) != 0)
+        {
+            auto const length = component.End - component.Start;
+            argument = new WCHAR[length + 1];
+            wcsncpy(argument, component.Start, length);
+            argument[length] = '\0';
+        }
+        return argument;
+    }
+
+    NewArrayHolder<WCHAR> _provider;
+    uint64_t _enabledKeywords;
+    uint32_t _level;
+    NewArrayHolder<WCHAR> _argument;
     bool _isValid;
 };
+#endif // FEATURE_PERFTRACING
+
+#if defined(HOST_UNIX) && (defined(FEATURE_EVENT_TRACE) || defined(FEATURE_EVENTSOURCE_XPLAT))
```

Comment:
@sywhang, we updated the build in #32746, so that user can build coreclr without `FEATURE_EVENT_TRACE` enabled. After rebasing my other PR https://github.com/dotnet/runtime/pull/32800 branch against master, I am getting errors that `XplatEventLoggerConfiguration` on line 408 is not defined. The reason is that `FEATURE_PERFTRACING` is only set when `FEATURE_EVENT_TRACE` is enabled, so the class definition is omitted from the compilation.

One fix could be to change line 245 to ` #if defined(FEATURE_PERFTRACING) || defined(FEATURE_EVENTSOURCE_XPLAT)` , (along with #32800 current changes), just wanted to confirm if you have any thoughts on such a fix?

Thanks!

Reply:
@am11 ah, sorry about that! 

> One fix could be to change line 245 to #if defined(FEATURE_PERFTRACING) || defined(FEATURE_EVENTSOURCE_XPLAT) , (along with #32800 current changes), just wanted to confirm if you have any thoughts on such a fix?

That sounds fine to me. 

I will try to build without FEATURE_EVENT_TRACE before merging my PRs on EventPipe from future to prevent issues like that :)

Reply:
Thank you! My coreclr build on Android was failing at 11% and with that (two liner) patch (updated #endif comment as well), it has just succeeded (all green). 👍 

I was wondering if we could get to run Android build even on weekly bases ([AzDo scheduled jobs](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml#scheduled-triggers)), that would help preserving the healthy state for this platform? It will also help validating this feature matrix. :)